vault/resource_identity_oidc_key_allowed_client_id.go

```diff
@@ -11,6 +11,7 @@ import (
 
 func identityOidcKeyAllowedClientId() *schema.Resource {
 	return &schema.Resource{
+		DeprecationMessage: "This resource is a less-featured dupe and is being deprecated in favor of the `vault_identity_oidc_key` resource. Please update your configuration.",
```

Comment:
Hey @tyrannosaurus-becks, this resource isn't really a duper per se. This is meant to allow the user to specify the `alllowed_client_ids` of an OIDC key in a decoupled manner.

We discussed this in https://github.com/terraform-providers/terraform-provider-vault/pull/488